### DISCUSSION TITLE
Embedding 3 or more IDEs in a HTML file using Theia

### DISCUSSION THREAD

**Discussion by maxloopinmok** [eclipse-theia/theia#12288] [General]
hi, is there a easy way to embed 3 or more IDEs inside a HTML page?  
```
        <div class="columns is-desktop">
          <div class="column is-3">
          <!-- First bottom quarter column -->
          <section class="section">
            <h1 class="title"><b>REPOSITORIES</b></h1>
          </section>
        </div>
        <div class="column is-3">
          <!-- Second bottom quarter column -->
          <section class="section">
            <h1 class="title"><b>Embedded Repo 1</b></h1>
          </section>
        </div>
        <div class="column is-3">
          <!-- Third bottom quarter column -->
          <section class="section">
            <h1 class="title"><b>Embedded Repo 2</b></h1>
          </section>
        </div>
        <div class="column is-3">
          <!-- Fourth bottom quarter column -->
          <section class="section">
            <h1 class="title"><b>Embedded Repo 3</b></h1>
          </section>
```

**Comment by msujew** [comment]
@maxloosmu The recommended way to embed Theia in another webpage is by using [`iframe`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe). Note that Theia also supports multi-root workspaces (i.e. workspaces which contain multiple repositories/directories). Maybe that would fit your requirements a bit better than three separate instances of Theia.

**Reply by maxloopinmok** [reply]
hi @msujew , thanks!  i've tried iframes, but although they can display, we can't really control what's going on in the iframe?  i've also tried multi-root workspaces, but they don't seem to work?  See my post at:
[https://community.theia-ide.org/t/how-to-open-3-or-more-tabs-in-a-theia-ide-in-a-html-file-programmatically/2874](https://community.theia-ide.org/t/how-to-open-3-or-more-tabs-in-a-theia-ide-in-a-html-file-programmatically/2874)
